### PR TITLE
only ramp latest canary revision

### PR DESCRIPTION
**Pull request by matkam**
Hello @dokipen, @ddbenson, @dnelson, @eblume, @silverlyra, @micahnoland, @tonymeng, 

Please review the following commits I made in branch 'matkam/only-release-latest-rev':

- **dont assume predictable ordering of revisions** (36a493b1b248042f0dccf55484a79e88775e81c9)

- **passing test and some refactoring** (d924cd34b32395219a7e9b35532cd323c7ba5192)

- **test prepareRevisionsAndRules (not passing)** (c238d41b3cf239a513c3e80e4e2f270ad4d445da)

- **mock incarnations** (373848f813b718d23e86799771ca372c1413d9ad)

- **regenerate mock deployment** (5ff4f7efffcd161fb18fefa9edb245915284222a)

- **missed a file** (d23007eca06da2aa071546e13e4cefcc698a5fee)

- **IncarnationInterface to Deployment** (f3f7674b83ce8327010434c82f3ac1b4acaa5a95)

- **incarnation interface** (de59e28140e5a18d5be6c23821b128e84b51b55d)

- **minor cleanup** (2a1f79857c466a309e2c8a7c5eb8f367fa614ec6)

- **bob is the greatest** (0831d65622579ef24d5741cac7b2f942e9aa3360)

- **mark older revisions as ineligible** (33d35e97bd82d611722264ae988eb8433ea2499c)

R=@dokipen
R=@ddbenson
R=@dnelson
R=@eblume
R=@silverlyra
R=@micahnoland
R=@tonymeng

### CODE REVIEW

File: pkg/controller/releasemanager/syncer_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+	var revisions []rmplan.Revision
+	for i := 0; i < 4; i++ {
+		revisions, _ = testResourceSyncer.prepareRevisionsAndRules()
+	}
+	t.Logf("revisions - %v", revisions)
+	logIncarnations(t, "deployed", deployedIncarnations)
+	logIncarnations(t, "releasableIncarnations", releasableIncarnations)
+
+	assert.Equal(t, uint32(100), releasableIncarnations[0].status.CurrentPercent)
```
Comment (dokipen) [inline pkg/controller/releasemanager/syncer_test.go]: shouldn't this be 50/20/30?

Comment (matkam) [reply pkg/controller/releasemanager/syncer_test.go]: Look at line 107 - I'm running `prepareRevisionsAndRules()` 4 times

Comment (matkam) [reply pkg/controller/releasemanager/syncer_test.go]: Does this mean "the releasables" should not include the currently deployed release? I'll update my tests, taking this into consideration.

Comment (matkam) [reply pkg/controller/releasemanager/syncer_test.go]: I just updated the test cases to test for both "bad" and "normal" releasable states

**Pull request merged**